components/raftstore/src/store/fsm/peer.rs

```diff
@@ -148,10 +149,15 @@ where
 {
     fn drop(&mut self) {
         self.peer.stop();
+        let mut raft_messages_size = 0;
         while let Ok(msg) = self.receiver.try_recv() {
             let callback = match msg {
                 PeerMsg::RaftCommand(cmd) => cmd.callback,
                 PeerMsg::CasualMessage(CasualMessage::SplitRegion { callback, .. }) => callback,
+                PeerMsg::RaftMessage(im) => {
+                    raft_messages_size += im.heap_size;
```

Comment:
I am afraid that sometimes here we would miss some messages. Could we use `Drop` to trace dropped messages?

Reply:
If `Drop` is implemented for `InspectedRaftMessage` we can't move things out from the message like this:
```rust
let InspectedRaftMessage { heap_size, mut msg } = msg;
```
And Implementing `Drop` for it requires to update `MEMTRACE_RAFT_MESSAGES` when it's constructed. To ensure this will make code more complex.

Reply:
Here the logic only called after a Region is destroyed. After this no messages can be sent to its channel. And all messages in its channel can be fetched in the `while` loop. So no messages can be missed theoretically.